src/architecture/unixfs.md

```diff
@@ -0,0 +1,618 @@
+
+If present, this field MUST be equal to the `Blocksize` computation above.
+Otherwise, this file is invalid.
+
+##### Path Resolution
+
+A file terminates a UnixFS content path. Any attempt to resolve a path past a
+file MUST error.
+
+#### `Directory` Type
+
+A :dfn[Directory], also known as folder, is a named collection of child :ref[Nodes]:
+
+- Every link in `PBNode.Links` is an entry (child) of the directory, and
+  `PBNode.Links[].Name` gives you the name of such child.
+- Duplicate names are not allowed. Therefore, two elements of `PBNode.Link` CANNOT
+  have the same `Name`. if two identical names are present in a directory, the
+  decoder MUST fail.
+
+The minimum valid `PBNode.Data` field for a directory is as follows:
+
+```json
+{
+  "Type": "Directory"
+}
+```
+
+The remaining relevant values are covered in [Metadata](#metadata).
+
+##### Link Ordering
+
+The canonical sorting order is lexicographical over the names.
+
+In theory there is no reason an encoder couldn't use an other ordering, however
+this lose some of its meaning when mapped into most file systems today (most file
+systems consider directories are unordered-key-value objects).
+
+A decoder SHOULD, if it can, preserve the order of the original files in however
+it consume those names. However when, some implementation decode, modify then
+re-encode some, the original links order fully lose it's meaning (given that there
+is no way to indicate which sorting was used originally).
+
+##### Path Resolution
+
+Pop the left-most component of the path, and try to match it to the `Name` of
+a child under `PBNode.Links`. If you find a match, you can then remember the CID.
+You MUST continue the search. If you find another match, you MUST error since
+duplicate names are not allowed. <!--TODO: check Kubo does this-->
+
+Assuming no errors were raised, you can continue to the path resolution on the 
+remaining components and on the CID you popped.
+
+
+#### `Symlink` type
+
+A :dfn[Symlink] represents a POSIX [symbolic link](https://pubs.opengroup.org/onlinepubs/9699919799/functions/symlink.html).
+A symlink MUST NOT have children. <!--TODO: check that this is true-->
+
+The `PBNode.Data.Data` field is a POSIX path that MAY be appended in front of the
+currently remaining path component stack.
+
+##### Path Resolution
+
+There is no current consensus on how pathing over symlinks should behave. Some
+implementations return symlink objects and fail if a consumer tries to follow them
+through.
+
+Following the POSIX specification over the current UnixFS path context is probably fine.
+
+#### `HAMTDirectory`
+
+A :dfn[HAMT Directory] is a [Hashed-Array-Mapped-Trie](https://en.wikipedia.org/wiki/Hash_array_mapped_trie)
+data structure representing a :ref[Directory]. It is generally used to represent
+directories that cannot fit inside a single block. They are also known as "sharded
+directories" since they allow to split large directories into multiple blocks, the "shards".
```

Comment:
```suggestion
A :dfn[HAMT Directory] is a [Hashed-Array-Mapped-Trie](https://en.wikipedia.org/wiki/Hash_array_mapped_trie)
data structure representing a :ref[Directory]. It is generally used to represent
directories that cannot fit inside a single block. These are also known as "sharded
directories:, since they allow you to split large directories into multiple blocks, known as "shards".
```